### ISSUE TITLE
refine "recently active" criteria

### ISSUE THREAD

**Issue by JonathanReeve** [mlaa/cbox-mla#186]
Members and groups are labeled as "recently active" when the activity is trivial--for members, logging in, and for groups, visiting that group. We should have a higher threshold for what constitutes a significant recent activity.

**Comment by JonathanReeve** [comment]
It looks like groups are showing up in the "recently active groups" widget if they've had recent membership changes, and members appear in "recently active members" if they've joined or left groups. It seems like what we need to do is prevent membership changes from updating that group or member's last activity.

**Comment by JonathanReeve** [comment]
We can do this by removing the call to `groups_update_groupmeta()` from the BP function `groups_join_group()`, but to do that we'd have to fork BuddyPress. The right thing to do, I think, would be to wrap the function call in an `apply_filters`, and then submit that upstream as a pull request. That way, other BP users might be able to limit their recent activity criteria, too. I'll write an issue report for this on BP Trac.

**Comment by JonathanReeve** [comment]
Submitted this as a bug report upstream at https://buddypress.trac.wordpress.org/ticket/6158. I'll see if I can put together a patch for it.

**Comment by JonathanReeve** [comment]
Patches submitted upstream; awaiting review from BP.

**Comment by JonathanReeve** [comment]
Putting this on hold awaiting an upstream merge. BP maintainers have tagged the above issue for 2.3, so hopefully it should be soon. We also have the option of forking BP if we want this fix now, but I'd vote to wait for it to float downstream first.

**Comment by JonathanReeve** [comment]
Still waiting on this. Removing the milestone in the meantime.

**Comment by terrainsvagues** [comment]
I'd like to push for forking BP if we can. This from a member today: :"The “last active” tracker just seems to track views – so groups that are highlighted as recently active actually have no content – the active people just lurked."

**Comment by JonathanReeve** [comment]
It's not a view that updates the most recent activity, as this member guessed, but a change in that group's membership. Since our custom plugin automatically subscribes people to groups when they log in, a group could experience a membership change when a member logs in for the first time.

Let's talk about this in our next meeting. If there's no other way to do it, we could fork BP, but it'd definitely be a Major Procedure with the potential to cause more problems than it solves, and it would  introduce a new layer of maintenance tasks. It would probably also require a new round of testing above and beyond the testing we're doing for the current release. If we were to do something like this, I'd slate it for Spy, since we're effectively in a feature freeze for Sentimental at the moment.

**Comment by JonathanReeve** [comment]
[Getting closer](https://buddypress.trac.wordpress.org/ticket/6158#comment:9)!

**Comment by JonathanReeve** [comment]
Fixed on dev (3.0). Buddypress 2.3.0 was released a few days ago. Now just waiting for CBOX. I'll leave this open for 2.X, since I imagine we'll want to backport this change, too, and we might want to wait for CBOX for that one.

**Comment by JonathanReeve** [comment]
Moving the milestone from 3.0 (where it is fixed) to 2.X (where it is still not fixed).

**Comment by rekmla** [comment]
Looks like this buddpress change was released in v 2.2.3.1 and is already on our system. We can address explicit membership changes now - without BP 2.3.x. However, the cbox-auth sync is setting last_activity too aggressively. I think this needs to be fixed in cbox-auth.
